src/main.rs

```diff
@@ -1,15 +1,8 @@
-use beacon_api_client::{Client, StateId};
-use url::Url;
+use beacon_api_client::{cli, CliArgs};
+use clap::Parser;
 
 #[tokio::main]
 async fn main() {
-    let s = "http://127.0.0.1:8003/";
-    let url: Url = Url::parse(s).unwrap();
-    let client = Client::new(url);
-
-    let checkpoints = client.get_finality_checkpoints(StateId::Finalized).await.unwrap();
-
-    println!("previous checkpoint: {:?}", checkpoints.previous_justified);
-    println!("current checkpoint: {:?}", checkpoints.current_justified);
-    println!("finalized checkpoint: {:?}", checkpoints.finalized);
+    let args = CliArgs::parse();
+    cli(args).await;
```

Comment:
I have a personal style that is more like
```suggestion
    let config = Config::parse();
    cli(&config).await;
```

I'd also pass the data by ref (see that `&`), rather than by value here